Initial commit

Believerin opened the pull request.

Believerin committed:
Initial commit

Believerin committed:
Initial commit

Believerin committed:
Initial commit

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/user/dto/UserDto.java:
```diff
@@ -0,0 +1,14 @@
+@AllArgsConstructor
+public class UserDto {
+    int id;
+    String name;
```
В дто тоже можно поставить аннотации для валидации полей

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,62 @@
+
+    @Override
+    public User addUser(User user) {
+        if (userStorage.findAll().stream().anyMatch(o -> o.getEmail().equals(user.getEmail()))) {
```
т.е. для того, что бы проверить есть ли пользователь в таблице тебе приходится вытащить всю таблицу юзеров, потом ее перебрать. Это не очень рационально (а если там миллионы пользователей).
Лучше сделать запрос, который это сразу и проверит

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,62 @@
+    public UserDto getUser(int userId) {
+        User user = userStorage.getUser(userId);
+        if (user == null) {
+            throw new NoSuchBodyException("Запрашиваемый пользователь");
```
Не совсем понятно сообщение об исключении

Believerin replied:
`NoSuchBodyException` может быть и у `Item`, и у `User`. В классе обработки исключений имеется общая для обоих сущностей фраза:
`return new ErrorResponse(String.format("%s отсутствует или не найден", e.getParameter()));`

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/user/dao/UserDbStorage.java:
```diff
@@ -0,0 +1,39 @@
+
+import java.util.*;
+
+@Component
```
Можно поставить @Repository

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/item/dao/ItemDbStorage.java:
```diff
@@ -0,0 +1,55 @@
+                if (i.getId() == itemId) return i;
+            }
+        }
+        return null;
```
Возвращать null там, где его не ждут чревато NPE. Наверное лучше какое то исключение выбросить. И в других методах тоже

Believerin replied:
Добавил в `ItemServiceImpl` обработку возвращаемого `null`

Alexander-Bolotov replied:
Ок, я приму это, но вдруг кто-то другой захочет воспользоваться твоим методом. Из названия не понятно, что он может вернуть нул, например

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/request/ItemRequest.java:
```diff
@@ -1,7 +1,22 @@
+@EqualsAndHashCode
+@RequiredArgsConstructor
+@AllArgsConstructor
+//@NoArgsConstructor(force=true)
```
Если код не нужен - удаляй

Alexander-Bolotov commented on src/main/java/ru/practicum/shareit/request/ItemRequest.java:
```diff
@@ -1,7 +1,22 @@
+@AllArgsConstructor
+//@NoArgsConstructor(force=true)
 public class ItemRequest {
+    int id; //уникальный идентификатор запроса
```
Излишние комментарии только мешают чтению кода. Достаточно понятных имен переменных

Believerin committed:
perf: "проверка на дублирование почты изменена и перенесена в UserDbStorage"
fix: "в метод getItem добавлена обработка null"
feat: "в dto добавлены аннотации валидации"
feat: "аннотация @Component заменёна на @Repository"
